internal/utils/utils.go

```diff
@@ -0,0 +1,51 @@
+package utils
+
+import (
+	"errors"
+	"math"
+)
+
+func SliceToChanks(sliceIn []int, chankSize int) [][]int {
+	if sliceIn == nil ||chankSize >= len(sliceIn) || chankSize < 1{
+		res := make([][]int,1)
+		res[0] = sliceIn
+		return res
+	} else {
+		numChanks := int(math.Ceil(float64(len(sliceIn)) / float64(chankSize)))
+		res := make([][]int,numChanks)
+		k := -1
+		for i:=0;i<len(sliceIn);i++ {
+			if i % chankSize == 0 {
+				k++
+				res[k] = make([]int,0)
+			}
+			res[k] = append(res[k],sliceIn[i])
+		}
+		return res
+	}
+}
+
+func InverseMap(mapIn *map[int]string) (*map[string]int,error){
+	res := make(map[string]int)
+	for key,val := range *mapIn {
+		if _,ok := res[val];ok {
+			return nil, errors.New("cant inverse, not uniq values at map")
+		}
+		res[val] = key
+	}
+	return &res,nil
+}
+
+func SliceFilter(sliceIn []int, filter []int) []int {
+	tmpset := make(map[int]bool)
```

Comment:
Тут не нужно исправлять, просто в коде можно встретить такое:
map[int]struct{} - что означает пустую структуру вместо bool -  тут особенность, что она не занимает память. Ее так же отправляют через каналы и тд

FYI:
`The struct type with no fields is called the empty struct, written struct{}. It has size zero and carries no information but may be useful nonetheless. Some Go programmers use it instead of bool as the value type of a map that represents a set, to emphasize that only the keys are significant, but the space saving is marginal and the syntax more cumbersome, so we generally avoid it.`

Reply:
Да, как раз читал про два способа организации сетов. Пустая структура используется по умолчанию?

Reply:
Просто 2 подхода) использовать нужно руководствуясь просто удобством и читаемостью кода